server/ai_process.go

```diff
@@ -1495,9 +1495,14 @@ func processAIRequest(ctx context.Context, params aiRequestParams, req interface
 	for {
 		select {
 		case <-cctx.Done():
-			err := fmt.Errorf("no orchestrators available within %v timeout", processingRetryTimeout)
-			if monitor.Enabled {
-				monitor.AIRequestError(err.Error(), monitor.ToPipeline(capName), modelID, nil)
+			err := cctx.Err()
+			if errors.Is(err, context.DeadlineExceeded) {
+				err = fmt.Errorf("no orchestrators available within %v timeout", processingRetryTimeout)
+				if monitor.Enabled {
+					monitor.AIRequestError(err.Error(), monitor.ToPipeline(capName), modelID, nil)
+				}
+			} else {
+				err = fmt.Errorf("context cancelled: %v", err)
```

Comment:
Won't this become `context cancelled: context canceled`? Perhaps check if the error is `context.Canceled` first before wrapping 🤔

Reply:
You're right. let's not wrap at all.